.github/workflows/lint-and-test.yml

```diff
@@ -13,6 +13,13 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Fetch main and PR history
+        run: |
+          git pull --unshallow origin ${{ github.head_ref }}
+          git fetch origin main
```

Comment:
better to use `depth: 0` instead?

Reply:
I think this is technically more efficient since it only downloads the base branch at max depth instead of every branch at max depth.